src/js/core.js

```diff
@@ -351,7 +355,7 @@
             element.setAttribute('data-medium-editor-element', true);
             element.setAttribute('role', 'textbox');
             element.setAttribute('aria-multiline', true);
-            element.setAttribute('medium-editor-index', index);
+            element.setAttribute('medium-editor-index', this.guid());
```

Comment:
I'm a bit concerned about replacing the `medium-editor-index` attribute directly with a guid for backward compatibility...

I saw you remove a `medium-editor-uid` attribute whenever the editor is destroyed but I don't see anywhere in the code where that attribute is created.  Were you planning to set the guid as `medium-editor-uid`?  Could you do that instead of changing the value of `medium-editor-index`?  If there isn't a need for `medium-editor-index` moving forward, we can look to remove it in the next major release (6.0.0).